internal/provider/node.go

```diff
@@ -0,0 +1,203 @@
+				Architecture:            runtime.GOARCH,
+				ContainerRuntimeVersion: system.Version(),
+				KernelVersion:           system.Kernel(),
+				KubeletVersion:          opts.Version,
+				OperatingSystem:         DefaultOperatingSystem,
+				OSImage:                 system.Image(),
+			},
+		},
+	}, nil
+}
+
+// computeNodeAddresses finds the internal and external address of the node (if found).
+func computeNodeAddresses(config *Opts) ([]corev1.NodeAddress, error) {
+	// Profile CIDRs reserved for private subnets.
+	// This is useful when auto-detecting IPs and segregating which are private.
+	cidrs := []string{"10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16"}
+	rfc1918 := make([]*net.IPNet, len(cidrs))
+	for i, cidr := range cidrs {
+		_, block, _ := net.ParseCIDR(cidr)
+		rfc1918[i] = block
+	}
+	// Try and auto-detect node addresses.
+	var internalIPs, externalIPs []string
+	for _, ip := range system.IPs() {
+		for _, block := range rfc1918 {
+			// Is internal IP?
+			if block.Contains(ip) {
+				internalIPs = append(internalIPs, ip.String())
+				continue
+			}
+
+			externalIPs = append(externalIPs, ip.String())
+		}
+	}
+
+	var nodeAddresses []corev1.NodeAddress
+	var errs []error
+
+	// If specified IP is 0.0.0.0, return auto-detected internal IPs.
+	if config.NodeInternalIP.IsUnspecified() {
+		if len(internalIPs) == 0 {
+			errs = append(errs, fmt.Errorf("failed to auto-detect internal IP"))
+		} else {
+			for _, ip := range internalIPs {
+				nodeAddresses = append(nodeAddresses, corev1.NodeAddress{Address: ip, Type: corev1.NodeInternalIP})
+			}
+		}
+		// Otherwise, return pre-defined internal IP.
+	} else {
+		nodeAddresses = append(nodeAddresses, corev1.NodeAddress{Address: config.NodeInternalIP.String(), Type: corev1.NodeInternalIP})
+	}
+
+	// If specified IP is 0.0.0.0, return auto-detected external IPs.
+	if config.NodeExternalIP.IsUnspecified() {
+		if len(externalIPs) == 0 {
+			errs = append(errs, fmt.Errorf("failed to auto-detect external IP"))
+		} else {
+			for _, ip := range externalIPs {
+				nodeAddresses = append(nodeAddresses, corev1.NodeAddress{Address: ip, Type: corev1.NodeExternalIP})
+			}
+		}
+		// Otherwise, return pre-defined external IP.
+	} else {
+		nodeAddresses = append(nodeAddresses, corev1.NodeAddress{Address: config.NodeExternalIP.String(), Type: corev1.NodeExternalIP})
+	}
+
+	return nodeAddresses, nil
+}
+
+// capacity returns a resource list containing the identifiable system resources.
+func capacity() corev1.ResourceList {
+	return corev1.ResourceList{
+		"cpu":     resource.MustParse(system.CPU()),
+		"memory":  resource.MustParse(system.Memory()),
+		"pods":    resource.MustParse(system.Pid()),
+		"storage": resource.MustParse("40G"), // needs the size of the /var FS
+	}
+}
+
+func nodeConditions() []corev1.NodeCondition {
+	return []corev1.NodeCondition{
+		// Don't set the Ready condition to True just yet.
+		// That shall happen when all of systemk has been initialized.
```

Comment:
can we point to a go file/function where that happens?

Reply:
It's in root.go, as the last step after Node and Pod controllers are running. https://github.com/virtual-kubelet/systemk/pull/65/files#diff-ab967ab1a2f3a1b769106eeb7bfe892ef0e81d1d27811fa15be08e6749feee1fR209

Reply:
please add this to the comment

Reply:
Done!